### ISSUE TITLE
Uncaught (in promise) DOMException: Failed to load because no supported source was found.

### ISSUE THREAD

**Issue by freeall** [feross/webtorrent#849]
* WebTorrent version: 0.95.2
* Node.js version: 5.10.1

I think `.appendTo` is currently broken and it doesn't correctly call its callback when loading errors occur. I use it like this:
``` js
file.appendTo('body', function (err, elem) {
  console.log('appendTo', err, elem)
})
```

But my callback is never called and I get this instead:
![image](https://cloud.githubusercontent.com/assets/376654/16506651/19643f5c-3f24-11e6-8eb2-27af51cb83d2.png)

(note that there's no stacktrace)

The result is that it's not possible to give a good error message to the user when loading errors happen. Maybe it's a problem in a subdependency?

**Comment by freeall** [comment]
Hmm, wait. It appears to just happen very slowly at times.
I was confused because the element was actually visible. I'll try to hide it until the callback is called instead.

**Comment by feross** [comment]
Ah, this is happening because the render functionality is trying multiple methods of playing back the video:

- `videostream` -- best, supports streaming *with* seeking, only works with MP4 for now
- `mediasource` -- supports more formats, supports streaming, but *no seeking*
- `blob` -- supports the most formats of all, with seeking, but *no streaming* (entire file must be downloaded first)

What's happening is that your media file is falling back to the blob approach, so it's waiting for a long time -- downloading potentially 100s of MBs before it realizes it can't play the format.

I just opened an issue on `render-media` to set a sane cap: https://github.com/feross/render-media/issues/18

**Comment by feross** [comment]
This is fixed in `render-media` 2.7.0. Now, the blob url strategy will not be attempted if the file is larger than 100MB, so you'll get an error in a more reasonable amount of time.

You can re-install `webtorrent` to get the fix.
